streamer/src/recvmmsg.rs

```diff
@@ -62,8 +71,8 @@ pub fn recv_mmsg(sock: &UdpSocket, packets: &mut [Packet]) -> io::Result<(usize,
         hdrs[i].msg_hdr.msg_iovlen = 1;
     }
     let mut ts = timespec {
-        tv_sec: 1,
-        tv_nsec: 0,
+        tv_sec: (timeo_ms / 1_000) as i64,
```

Comment:
What is the benefit of taking the timeout here? I presume it would lower it from 1s to 1ms or 5ms for all cases. It would go into recv_mmsg more times, of course once per 1ms is much better than 500 per 1ms.

In the blocking case, it would already return if there are packets. And once there are packets and it goes into non-blocking it would ignore this, is that right?

Reply:
Yes... I'm also curious about the original motivation for switching from blocking to non-blocking when the callers are spinning on this routine until the timeout is reached or the buffer is full. If we specify a timeout and leave the call blocking it should return when either the timeout is reached or the buffer is full. Do you see some other reason for this?

Reply:
I found this, the timeout behavior for recvmmsg is somewhat misleading.